docs/SECURITY.md

```diff
@@ -0,0 +1,316 @@
+clean way.
+
+**Synchronous Network**: A packet sent from an alive node reaches its
+destination in a bounded amount of time. Drand realizes this assumptions by the
+usage of timeouts during the DKG protocol.
+
+**Synchronized Clocks**: All nodes must have roughly synchronized clocks (less
+than one 1s of drift).
+
+**Reliable Broadcast Channel**: When a node broadcasts a packet to all other
+nodes, each other node is guaranteed to receive the same exact packet after some
+bounded amount of time. This assumption is not strictly realized by drand
+currently. See [DKG attacks](#dkg-attacks) section to understand the impact.
+
+**Authenticated Channel**: Every communication between nodes must be
+authenticated. Drand achieves this by signing every outgoing DKG packets with
+a BLS signature on the longterm public key of the sender node.
+
+**Public Group**: Every nodes willing to run the DKG must know the group
+formation before starting the DKG, including the longterm public keys of each
+node. During the DKG, there might be some nodes offline or misbehaving. The set
+of nodes that _successfully_ passed the DKG are called the _qualified_ set of
+nodes (QUAL). Only these nodes have valid shares and are able to produce
+partial beacons that can be successfully validated with respect to the
+distributed public key.
+
+**DKG's biasability in signatures**: Pedersen's DKG is known to exhibit a
+weakness in the biasability in the distribution of the distributed private key.
+However, the same authors (Gennaro et al.) that proved the latter also proved
+this bias is not relevant in the setting of using the DKG to perform digital
+signatures, which offers other strong properties -
+[paper](https://pdfs.semanticscholar.org/642b/d1bbc86c7750cef9fa770e9e4ba86bd49eb9.pdf).
+In particular, the paper mentions discrete log based systems. However, it is not
+yet strictly proven that system using computation co-CDH assumptions as required
+for threshold BLS signatures are secure in the model of Gennaro.  However, it is
+believed that this assumptions holds in this context as well and is being worked
+on.  Note that using threshold BLS signatures as a source of randomness _is_
+formally proven secured in this [paper](https://eprint.iacr.org/2020/096.pdf)
+from Galindo et al. 
+
+### Randomness generation model
+
+**Network**: The randomness generation protocol do not make any assumption on
+the network bounds. As soon as packet comes in, node processes them and the
+chain advances if conditions are there (enough partial beacon and time for a new round).
+
+**Synchronized Clocks**: All nodes must have roughly synchronized clocks to
+start the rounds at the same time. The precision of the synchronicity between
+clocks only needs to be at the order the round frequency (order of tens of
+seconds), which is much higher than the reality of server's clock (NTP-synced
+servers have drifts of under a second over the globe). XXX need source.
```

Comment:
```suggestion
seconds), which is much higher than the reality of the server's clock (NTP-synced
servers achieve [offsets of under a second over the globe](https://www.eecis.udel.edu/~mills/database/reports/ntp-survey99-minar.pdf)). 
```
This is a fairly standard source, even though pretty old. It's a pretty uncontroversial statement, though -- NTP can generally provide 10s of ms anywhere around the world because there are local (low-latency) stratum 1 servers in every country and in most networks.